src/main/java/org/prebid/cache/handlers/PostCacheHandler.java

```diff
@@ -86,7 +86,8 @@ public PostCacheHandler(final ReactiveRepository<PayloadWrapper, String> reposit
         val payloadFlux = flux
                 .map(payload -> payload.toBuilder()
                         .prefix(config.getPrefix())
-                        .expiry(adjustExpiry(payload.getExpiry()))
+                        .expiry(payload.getTtlseconds() != null ? adjustExpiry(payload.getTtlseconds())
```

Comment:
1. `adjustExpiry(payload.getTtlseconds() != null ? payload.getTtlseconds() : payload.getExpiry())` looks a bit more concise to me.
2. It would be good to have a test for this logic.